eslint.config.mjs

```diff
@@ -0,0 +1,234 @@
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+import { fixupConfigRules, fixupPluginRules } from '@eslint/compat';
+import { FlatCompat } from '@eslint/eslintrc';
+import js from '@eslint/js';
+import tsParser from '@typescript-eslint/parser';
+import vitest from '@vitest/eslint-plugin';
+import etc from 'eslint-plugin-etc';
+import fileProgress from 'eslint-plugin-file-progress';
+import importPlugin from 'eslint-plugin-import';
+import noNull from 'eslint-plugin-no-null';
+import redundantUndefined from 'eslint-plugin-redundant-undefined';
+import simpleImportSort from 'eslint-plugin-simple-import-sort';
+import sonarjs from 'eslint-plugin-sonarjs';
+import svelte from 'eslint-plugin-svelte';
+import unicorn from 'eslint-plugin-unicorn';
+import globals from 'globals';
+import svelteParser from 'svelte-eslint-parser';
+import typescriptLint from 'typescript-eslint';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
+  allConfig: js.configs.all,
+});
+
+export default [
+  {
+    ignores: ['**/dist/**/*', '**/.svelte-kit/**', '**/coverage/**', '**/node_modules/**'],
+  },
+  js.configs.recommended,
+  ...typescriptLint.configs.recommended,
+  sonarjs.configs.recommended,
+  ...svelte.configs['flat/recommended'],
+  ...fixupConfigRules(
+    compat.extends('plugin:import/recommended', 'plugin:import/typescript', 'plugin:etc/recommended'),
+  ),
+  {
+    plugins: {
+      // compliant v9 plug-ins
+      unicorn,
+      'file-progress': fileProgress,
+      // non-compliant v9 plug-ins
+      etc: fixupPluginRules(etc),
+      import: fixupPluginRules(importPlugin),
+      'no-null': fixupPluginRules(noNull),
+      'redundant-undefined': fixupPluginRules(redundantUndefined),
+      'simple-import-sort': fixupPluginRules(simpleImportSort),
+      vitest,
+    },
+    settings: {
+      'import/resolver': {
+        typescript: true,
+        node: true,
+
+        'eslint-import-resolver-custom-alias': {
+          alias: {
+            '/@': './src',
+            '/@gen': './src-generated',
```

Comment:
`src-generated` ? we do not have generated stuff here ?

Reply:
removed